src/database/schema/ChannelListSchema.ts

```diff
@@ -514,14 +514,17 @@ class ChannelList implements BaseDbSchema {
     });
   }
 
-  static fromInitAnonymousChatAPI(data: ModifyAnonymousChatData): ChannelList {
+  static fromInitAnonymousChatAPI(
+    data: ModifyAnonymousChatData,
+    type: 'ANON_PM' | 'PM'
+  ): ChannelList {
     return new ChannelList({
       id: data?.message?.cid,
-      channelPicture: '',
+      channelPicture: data.targetImage,
       name: data?.targetName,
       description: data?.message?.message,
       unreadCount: 0,
-      channelType: 'ANON_PM',
+      channelType: type,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are good for flexibility and customization. Now, the `fromInitAnonymousChatAPI` method can accept a `type` parameter to set the `channelType`, and it uses `data.targetImage` to set the `channelPicture` instead of an empty string. This allows more dynamic creation of `ChannelList` instances based on the provided data. However, please ensure that the `type` parameter is always provided when calling this function, and `data.targetImage` is not null or undefined, otherwise, it may lead to unexpected behavior.
```diff
-      channelPicture: '',
+      channelPicture: data.targetImage,
-      channelType: 'ANON_PM',
+      channelType: type,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->